core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/UniqueIDUserStoreManager.java

```diff
@@ -390,4 +391,186 @@ void updateUserListOfRoleWithID(String roleName, String[] deletedUserIDs, String
      */
     Map<String, List<String>> getRoleListOfUsersWithID(List<String> userIDs) throws UserStoreException;
 
+    /**
+     * Retrieves list of group IDs of a given user ID.
+     *
+     * @param userId    User ID.
+     * @param limit     No of search results. If the given value is greater than the system configured max limit
+     *                  it will be reset to the system configured max limit.
+     * @param offset    Start index of the user search.
+     * @param sortBy    Sorted by.
+     * @param sortOrder Sorted order.
+     * @return List of Group objects.
+     * @throws UserStoreException thrown by the underlying UserStoreManager.
+     */
+    List<Group> getGroupListOfUser(String userId, int limit, int offset, String sortBy, String sortOrder)
+            throws UserStoreException;
+
+    /**
+     * Retrieves list of group IDs evaluating the condition.
+     *
+     * @param condition Conditional filter.
+     * @param limit     No of search results. If the given value is greater than the system configured max limit
+     *                  it will be reset to the system configured max limit.
+     * @param offset    Start index of the user search.
+     * @param sortBy    Sorted by.
+     * @param sortOrder Sorted order.
+     * @return
+     */
+    List<Group> getGroupList(Condition condition, int limit, int offset, String sortBy, String sortOrder)
+            throws UserStoreException;
+
+    /**
+     * Retrieves list of Users that belongs to a given group ID.
+     *
+     * @param groupID   Group ID.
+     * @param limit     No of search results. If the given value is greater than the system configured max limit
+     *                  it will be reset to the system configured max limit.
+     * @param offset    Start index of the user search.
+     * @param sortBy    Sort by.
+     * @param sortOrder Sort order.
+     * @return List of Users.
+     * @throws UserStoreException thrown by the underlying UserStoreManager.
+     */
+    List<User> getUserListOfGroup(String groupID, int limit, int offset, String sortBy, String sortOrder)
+            throws UserStoreException;
+
+    /**
+     * Retrieves list of Users by evaluating the condition.
+     *
+     * @param condition Conditional filter.
+     * @param limit     No of search results. If the given value is greater than the system configured max limit
+     *                  it will be reset to the system configured max limit.
+     * @param offset    Start index of the user search.
+     * @param sortBy    Sort by.
+     * @param sortOrder Sort order.
+     * @return List of Users.
+     * @throws UserStoreException thrown by the underlying UserStoreManager.
+     */
+    List<User> getUserList(Condition condition, int limit, int offset, String sortBy, String sortOrder)
```

Comment:
A similar method is already there in the interface. Check whether this is really needed.